picocrt/machine/arm/crt0.c

```diff
@@ -132,6 +155,40 @@ _start(void)
 	__asm__("vmsr fpexc, %0" : : "r" (0x40000000));
 #endif
 
+#ifdef PICOCRT_ENABLE_CACHES
+#define SCTLR_MMU (1 << 0)
+#define SCTLR_DATA_L2 (1 << 2)
+#define SCTLR_BRANCH_PRED (1 << 11)
+#define SCTLR_ICACHE (1 << 12)
+    /* We have to set up an identity map and enable the MMU for caches.
+     * Additionally, all page table entries are set to Domain 0, so set up DACR
+     * so that Domain zero has permission checks enabled rather than "deny all".
+     */
+    __asm__(
```

Comment:
Any trouble running this code on targets without the relevant hardware? If so, can we run-time detect it?

Reply:
QEMU does not have caches and is happy with this, but I haven't actually run it on any real core without cache.